### ISSUE TITLE
Devices support scrcpy -m2048

### ISSUE THREAD

**Issue by tulasikrishan89** [NetrisTV/ws-scrcpy#51]
some devices support working only if mentioned as --

scrcpy -m2048

How to handle this situation in ws-scrcpy. can it do it automatically?

**Comment by drauggres** [comment]
Hi.

TL;DR ws-srcpy automatically tries to fit the image into your browser window.

You can try to change it here: https://github.com/NetrisTV/ws-scrcpy/blob/369c6cf83185c72742224faef24172a1d754e0d5/src/server/Constants.ts#L6

But actually ws-scrcpy client receives this value (along with others: [VideoSettings](https://github.com/NetrisTV/ws-scrcpy/blob/master/src/app/VideoSettings.ts)) but uses them only if they are smaller than client (browser) window. Otherwise (or if there is only one receiving client at the moment) ws-scrcpy will send your browser window size as [bounds](https://github.com/NetrisTV/ws-scrcpy/blob/369c6cf83185c72742224faef24172a1d754e0d5/src/app/client/ScrcpyClient.ts#L100).

You can always change "max-size" on the fly in `More` menu (three dots) under `Change video settings` spoiler. Last manually changed settings are saved into local storage.

**Comment by drauggres** [comment]
In v0.7.0 you can set video settings (including `-m/--max-size` analogs `Max Width`&`Max Height`) before stream start.
